search-app/src/cmr/search/api/autocomplete.clj

```diff
@@ -0,0 +1,63 @@
+(ns cmr.search.api.autocomplete
+  "Defines the API for autocomplete-suggest in the CMR."
+  (:require
+    [clojure.string :as str]
+    [cmr.common.log :refer (debug info warn error)]
+    [cmr.common.services.errors :as svc-errors]
+    [cmr.search.services.autocomplete-service :as ac]
+    [cmr.search.services.query-service :as qs]
+    [cmr.common-app.api.routes :as common-routes]
+    [cmr.common.mime-types :as mt]
+    [compojure.core :refer :all]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Support Functions
+
+(def limit-default 20)
+(def offset-default 0)
+
+(defn- string-param-or-default->int
+  [s default]
+  (if s
+    (Integer/parseInt s 10)
+    default))
+
+(defn- get-autocomplete-suggestions
+  "Invokes Elasticsearch for autocomplete query result"
+  [ctx params]
+  (let [query-param (:q params)
+        type-param (:types params)]
+    (if query-param
+      (let [term (str/trim (str/lower-case query-param))
+            types (if type-param
```

Comment:
change `if` to `when` as you only have the `true` case.

Reply:
The false case is the 400 error below

Reply:
Kind of a nit but i think for readability it would be better to throw that validation towards the top of the function and have just `(when-not query-params <throw-exception-here>`. Or even split that out to its own function called `validate-query-params` or something.

Reply:
addressed in  b474eaf